### PR TITLE
systemd-cryptenroll: add page

### PR DESCRIPTION
**Pull request by RoepLuke**
Hello there, I stumbled upon systemd-cryptenroll today and when trying to understand it, I found it had no TLDR yet.

~~I tried to make it as compact as possible, but the very dynamic parameter set of this command did not allow me to compress it further without leaving out important points. Furthermore, I had to use H2-Headings to distance the different Unlocking Methods from each other.~~

~~I wasn't able to find a relatable / about as complex example to base this one on, but I would be inclined to adapt this pull request if you could provide me with one.~~ 

Also, ~~the Heading/Introduction had to be larger than normal~~ because of the important distinction from the system boot behavior defined in crypttab and to tell the User that the volume needs to be unlocked by systemd-cryptenroll via a currently present Method to make any changes.

~~**EDIT / TODO: I read the content guidelines a bit too late and need to slim the tldr before it can be reviewed and accepted.**~~

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** systemd 253 (253.9-1.fc38)

### CODE REVIEW

**Comment by RoepLuke** [comment]
I'd propose removing all semi-duplicate "Unlock with FIDO2-Device" and "Unlock with Key-File" examples.
They **could** be replaced by a single general example for using FIDO2/KeyFile to Unlock changing the volume Key Slots.
Or they could just be left out. That would cut the examples down from 28 to about 19.

`systemd-cryptenroll --unlock-fido2-device={{path/to/fido_hidraw_unlock_device}} --fido2-device={{path/to/fido_hidraw_enroll_device}} {{path/to/luks2_block_device}}` is kind of an exception here because it shows the difference between the two "fido2-device" parameters.

**Comment by RoepLuke** [comment]
Removing all semi-duplicate `[...] --wipe-slots= [...]` examples would cut them down from 19 to 15.

**Comment by RoepLuke** [comment]
The examples using `[...] --tpm2-device=list [...]` or `[...] --fido2-device=list [...]` or `[...] --pkcs11-token-uri=list [...]` could be combined with their other respective examples. 
Like this:
`systemd-cryptenroll --pkcs11-token-uri={{list|auto|pkcs11_token_uri}} {{path/to/luks2_block_device}}`

That would remove another 3 examples, bringing us down to 12. An now I'm not sure what to cut next.
Open for approval / disapproval / feeedback.

**Comment by RoepLuke** [comment]
The page/file has been updated to only contain 8 examples and its main description shortened. 
The descriptions of the examples may still be considered too long, but I'm not sure on that.

grafik

**Review by blueskyson — APPROVED** [review]

File: pages/linux/systemd-cryptenroll.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# systemd-cryptenroll
+
+> Interactively enroll or remove methods used to unlock LUKS2-encrypted partitions / block-devices.
```
Comment (kbdharun) [inline pages/linux/systemd-cryptenroll.md:3]:
```suggestion
> Interactively enroll or remove methods used to unlock LUKS2-encrypted partitions/block devices.
```

File: pages/linux/systemd-cryptenroll.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+> In order to allow a partition to be unlocked during system boot using something other than a Password, also update the crypttab file and initramfs.
+> More information: <https://www.freedesktop.org/software/systemd/man/systemd-cryptenroll.html>.
+
+- Unlock using Password, and enroll a new / additional Password:
```
Comment (kbdharun) [inline pages/linux/systemd-cryptenroll.md:7]:
```suggestion
- Unlock using Password, and enroll a new/additional Password:
```

File: pages/linux/systemd-cryptenroll.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+`systemd-cryptenroll --pkcs11-token-uri={{list|auto|pkcs11_token_uri}} {{path/to/luks2_block_device}}`
+
+- Unlock using Pasword, and list available or enroll a new FIDO2-Device (using PIN and Presence/Touch if available):
+
+`systemd-cryptenroll --fido2-device={{list|auto|path/to/fido2_hidraw_device}} {{path/to/luks2_block_device}}`
+
+- Unlock using Pasword, and enroll a new FIDO2-Device with User Vertification (Biometrics):
+
+`systemd-cryptenroll --fido2-device={{auto|path/to/fido2_hidraw_device}} --fido2-with-user-verification=yes {{path/to/luks2_block_device}}`
```
Comment (kbdharun) [inline pages/linux/systemd-cryptenroll.md:25]:
```suggestion
- Unlock using Password, and enroll a new/additional Recovery Key:

`systemd-cryptenroll --recovery-key {{path/to/luks2_block_device}}`

- Unlock using Password, and list available or enroll a new/additional PKCS#11 Token:

`systemd-cryptenroll --pkcs11-token-uri={{list|auto|pkcs11_token_uri}} {{path/to/luks2_block_device}}`

- Unlock using Pasword, and list available or enroll a new FIDO2-Device (using PIN and Presence/Touch if available):

`systemd-cryptenroll --fido2-device={{list|auto|path/to/fido2_hidraw_device}} {{path/to/luks2_block_device}}`

- Unlock using Password, and enroll a new FIDO2-Device with User Verification (Biometrics):

`systemd-cryptenroll --fido2-device={{auto|path/to/fido2_hidraw_device}} --fido2-with-user-verification=yes {{path/to/luks2_block_device}}`
```

File: pages/linux/systemd-cryptenroll.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+`systemd-cryptenroll --unlock-fido2-device={{path/to/fido2_hidraw_unlock_device}} --fido2-device={{path/to/fido2_hidraw_enroll_device}} {{path/to/luks2_block_device}}`
+
+- Unlock using Password, and enroll a TPM2 Security Chip (only secure-boot-policy PCR) and require additional alphanumeric PIN:
+
+`systemd-cryptenroll --tpm2-device={{auto|path/to/tpm2_block_device}} --tpm2-with-pin=yes {{path/to/luks2_block_device}}`
+
+- Unlock using Password, and remove all empty Passwords / all Passwords / all FIDO2-Devices / all PKCS#11 Tokens / all TMP2 Security Chips / all Recovery-Keys / all Methods:
+
+`systemd-cryptenroll --wipe-slots={{empty|password|fido2|pkcs#11|tpm2|recovery|all}} {{path/to/luks2_block_device}}`
```
Comment (kbdharun) [inline pages/linux/systemd-cryptenroll.md:37]:
```suggestion
- Unlock using a FIDO2-Device, and enroll a new FIDO2-Device:

`systemd-cryptenroll --unlock-fido2-device={{path/to/fido2_hidraw_unlock_device}} --fido2-device={{path/to/fido2_hidraw_enroll_device}} {{path/to/luks2_block_device}}`

- Unlock using Password, and enroll a TPM2 Security Chip (only secure-boot-policy PCR) and require additional alphanumeric PIN:

`systemd-cryptenroll --tpm2-device={{auto|path/to/tpm2_block_device}} --tpm2-with-pin=yes {{path/to/luks2_block_device}}`

- Unlock using Password, and remove all empty Passwords/all Passwords/all FIDO2-Devices/all PKCS#11 Tokens/all TMP2 Security Chips/all Recovery-Keys/all Methods:

`systemd-cryptenroll --wipe-slots={{empty|password|fido2|pkcs#11|tpm2|recovery|all}} {{path/to/luks2_block_device}}`
```

**Review by kbdharun — COMMENTED** [review]
Welcome to tldr and thanks for your contribution. I have some suggestions for this page.

We generally prefer to not leave spaces between `/` (slashes) across our pages.

**Review by kbdharun — APPROVED** [review]

**Pull request merged**